odxtools/standardlengthtype.py

```diff
@@ -82,53 +78,132 @@ def __get_raw_mask(self, internal_value: AtomicOdxType) -> Optional[bytes]:
             # been anticipated by the standard, though...
             endianness = "little"
 
+        if self.is_condensed:
+            # if a condensed bitmask is specified, the number of bits
+            # set to one in the bit mask are used in the PDU
+
+            # TODO: this is pretty slow. replace it by
+            # `self.bit_mask.bit_count()` once we require python >=
+            # 3.10.
+            bit_sz = bin(self.bit_mask).count("1")
+            used_mask = (1 << bit_sz) - 1
+
+            return used_mask.to_bytes((bit_sz + 7) // 8, endianness)
+
         sz: int
         if isinstance(internal_value, (bytes, bytearray)):
             sz = len(internal_value)
         else:
             sz = (odxrequire(self.get_static_bit_length()) + 7) // 8
 
         max_value = (1 << (sz * 8)) - 1
-        bit_mask = self.bit_mask & max_value
+        used_mask = self.bit_mask & max_value
 
-        return bit_mask.to_bytes(sz, endianness)
+        return used_mask.to_bytes(sz, endianness)
 
     def __apply_mask(self, internal_value: AtomicOdxType) -> AtomicOdxType:
         if self.bit_mask is None:
             return internal_value
+
         if self.is_condensed:
-            odxraise("Serialization of condensed bit mask is not supported", NotImplementedError)
-            return
+            int_value: int
+            if isinstance(internal_value, bytes):
```

Comment:
what about bytearray ?

Reply:
good point. fixed: https://github.com/mercedes-benz/odxtools/pull/382/commits/e4d7724855e77e89d0c259c5e4906358b4189a43

I'm not sure about the performance implications of `isinstance(..., SupportsBytes)` because I suppose that `SupportsBytes` is a type protocol and checking these is slow, but the vast majority of DOPs should not specify a bitmask, this should not matter too much...